tests/potential/test_potential.py

```diff
@@ -581,5 +581,211 @@ def test_saveload(self):
 
         temp.close()
 
+class QuadPot(potential.PairPotential):
+    """Quadratic potential function used to test potential.Tabulator"""
+
+    def __init__(self, types, params, default={}):
+        super().__init__(types, params, default)
+
+    def _energy(self, r, m, **params):
+        r,u,s = self._zeros(r)
+        u = m*(3-r)**2
+        if s:
+            u = u.item()
+        return u
+
+    def _force(self, r, m, **params):
+        r,f,s = self._zeros(r)
+        f = 2*m*(3-r)
+        if s:
+            f = f.item()
+        return f
+
+    def _derivative(self, param, r, **params):
+        r,d,s = self._zeros(r)
+        if param == 'm':
+            d = (3-r)**2
+        if s:
+            d = d.item()
+        return d
+
+class test_Tabulator(unittest.TestCase):
```

Comment:
If you make the change I proposed, these tests will need to be cleaned up a bit, so I will defer looking through them carefully until then. They look good overall though!